build-pipeline.yml

```diff
@@ -68,6 +68,10 @@ parameters:
   - name: ui_automation_ref
     type: string
     default: d621a86d52eb9d45704393a9d5cc02b30e836a8e
+  # Tells the build pipeline that there are resources to validate
+  - name: has_ui_resources
+    type: boolean
+    default: false
```

Comment:
Could we have default true instead of false to make sure that we check if the devs forgot about it?
It's the kind of check we are all going to forget pretty soon. So we need to make sure it always runs. Except for repos, we know it's pointless.
Does this make sense?

Reply:
In this case, the job should consider what will happen if it receives true and it doesn't have UI resources to build so it can skip instead of failing.